examples/reference/custom_components/CompositeWidget.md

```diff
@@ -0,0 +1,129 @@
+# `CompositeWidget`
+
+`CompositeWidget` simplifies the creation of custom Panel widgets using only Python and Panel components.
+
+```{pyodide}
+import param
+import panel as pn
+from panel.widgets import CompositeWidget
+
+pn.extension()
+
+class ButtonGroup(CompositeWidget):
```

Comment:
This component is really missing from Panel. It works in a way similar to https://www.gradio.app/docs/gradio/examples.

Its different from a `RadioButtonGroup` because 1) it does not show which option you selected last 2) instead you can reselect the same option and the `value` event will re-retrigger. This is very useful if the selection you made might be changed in other ways as seen in for example the second example.

Reply:
This component would also have been useful for @ahuang11 chat example in https://discourse.holoviz.org/t/panel-chat-interface/7505/5?u=marc. The current version does not have a great UX because the `RadioButton` used does not support an unselected state.